packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -380,6 +383,59 @@ class HierMergeBlock extends MergeBlock implements IHierBlock {
     clientId: string;
 }
 
+export interface IMergeTreeOptions {
+    catchUpBlobName?: string;
+    /**
+     * Whether to enable the length calculations implemented in
+     * https://github.com/microsoft/FluidFramework/pull/11678
+     * 
+     * These calculations resolve bugginess that causes eventual consistency issues in certain conflicting
+     * removal cases, but regress some index-based undo-redo implementations. The suggested path for
+     * consumers is to switch to LocalReference-based undo-redo implementation (see 
+     * https://github.com/microsoft/FluidFramework/pull/11899) and enable this feature flag.
+     * 
+     * default: false
+     */
+    mergeTreeUseNewLengthCalculations?: boolean;
+    mergeTreeSnapshotChunkSize?: number;
+    /**
+     * Whether to use the SnapshotV1 format over SnapshotLegacy.
+     * 
+     * SnapshotV1 stores a view of the merge-tree at the current sequence number, preserving merge metadata
+     * (e.g. clientId, seq, etc.) only for segment changes within the collab window.
+     * 
+     * SnapshotLegacy stores a view of the merge-tree at the minimum sequence number along with the ops between
+     * the minimum sequence number and the current sequence number.
+     * 
+     * Both formats merge segments where possible (see {@link ISegment.canAppend})
+     * 
+     * default: false
+     * 
+     * @remarks
+     * Despite the "legacy"/"V1" naming, both formats are actively used at the time of writing. SharedString
+     * uses legacy and Matrix uses V1.
+     */
+    newMergeTreeSnapshotFormat?: boolean;
+
+    /**
+     * Options related to attribution
+     */
+    attribution?: IMergeTreeAttributionOptions;
+}
+
+export interface IMergeTreeAttributionOptions {
+    /**
+     * If enabled, segments will store attribution keys which can be used with the runtime to determine
+     * attribution information (i.e. who created the content and when it was created).
+     * 
+     * This flag only applied to new documents: if a snapshot is loaded, whether or not attribution keys
+     * are tracked is determined by the presence of existing attribution keys in the snapshot.
+     * 
+     * default: false
+     */
+    track?: boolean;
```

Comment:
you may want to add a config for this as well, as it would allow you to turn it on at runtime manually and test it out, without having partners needing to modify code.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/2922